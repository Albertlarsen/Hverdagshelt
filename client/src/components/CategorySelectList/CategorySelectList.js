import { Component } from 'react';
import { CategoryService } from '../../services';
import * as React from 'react';
import { Alert } from '../../widgets';
import Select from 'react-select';

let categoryService = new CategoryService();


export class CategorySelectList extends Component<Props, State> {

  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      choosen: {},
      values: [
        {}
      ]
    };

    this.handleOnChangeCategory = this.handleOnChangeCategory.bind(this);
  }


  handleOnChangeCategory(e: Object) {
    this.props.handleOnChangeCategory(e.value, e.label);

  };

  componentWillMount() {
    var arr = [];
    categoryService
      .getCategory1()
      .then(cat1 => {
        cat1.map(e => {
          var elem = {
            name: e.name,
            categoryId: e.categoryId
          };
          arr = arr.concat(elem);

        });
        this.setState({
          values: arr
        });
      })
      .catch((error: Error) => Alert.danger(error.message));

  }

  render() {
    let optionTemplate = this.state.values.map(v => {
      var data = { label: v.name, value: v.categoryId };
      return (data);
    });
    return (
      <Select
        placeholder={'Kategorier'}
        name="colors"
        options={optionTemplate}
        className="basic"
        classNamePrefix="select"
        onChange={this.handleOnChangeCategory}
      />
    );
  }
}