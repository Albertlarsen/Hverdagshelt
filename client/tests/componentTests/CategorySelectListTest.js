// @flow

import {shallow} from "enzyme/build";
import {CategorySelectList} from "../../src/components/CategorySelectList/CategorySelectList";
import * as React from "react";

describe('Test for CategorySelectList component', () => {
    const wrapper = shallow(<CategorySelectList />);


    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

});