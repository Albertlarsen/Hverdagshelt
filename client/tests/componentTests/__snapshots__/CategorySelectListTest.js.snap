// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for CategorySelectList component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CategorySelectList />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "className": "basic",
      "classNamePrefix": "select",
      "defaultInputValue": "",
      "defaultMenuIsOpen": false,
      "defaultValue": null,
      "name": "colors",
      "onChange": [Function],
      "options": Array [
        Object {
          "label": undefined,
          "value": undefined,
        },
      ],
      "placeholder": "Kategorier",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "className": "basic",
        "classNamePrefix": "select",
        "defaultInputValue": "",
        "defaultMenuIsOpen": false,
        "defaultValue": null,
        "name": "colors",
        "onChange": [Function],
        "options": Array [
          Object {
            "label": undefined,
            "value": undefined,
          },
        ],
        "placeholder": "Kategorier",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
